src/algorithms/synthesis/dd_synthesis.cpp

```diff
@@ -169,13 +189,21 @@ namespace syrec {
     auto DDSynthesizer::swapPaths(dd::mEdge src, dd::mEdge const& current, TruthTable::Cube::Vector const& p1SigVec, TruthTable::Cube::Vector const& p2SigVec, std::unique_ptr<dd::Package<>>& dd) -> dd::mEdge {
         if ((current.p->e[0].isZeroTerminal() && !current.p->e[1].isZeroTerminal()) || (p2SigVec.size() > p1SigVec.size())) {
             TruthTable::Cube::Vector rootSigVec;
-            pathFromSrcDst(src, current, rootSigVec);
+            pathFromSrcDst(src, current.p, rootSigVec);
+
+            const auto rootSolution = esopOptimization(rootSigVec);
 
             const auto nQubits = static_cast<dd::QubitCount>(src.p->v + 1);
-            for (auto const& rootVec: rootSigVec) {
+
+            if (rootSolution.empty()) {
+                dd::Controls ctrlFinal1;
+                applyOperation(nQubits, current.p->v, src, ctrlFinal1, dd);
```

Comment:
```suggestion
                applyOperation(nQubits, current.p->v, src, {}, dd);
```
Not 100% sure if this works though.